src/orchestrator-process/src/lib.rs

```diff
@@ -135,7 +136,11 @@ impl NamespacedOrchestrator for NamespacedProcessOrchestrator {
                     .ok_or_else(|| anyhow!("port exhaustion"))?;
                 ports.insert(port.name.clone(), p);
             }
-            let args = args(&ports);
+            hosts_ports.push(("localhost".to_string(), ports))
+        }
+        for i in 0..(scale_in.get()) {
+            let ports = &hosts_ports[i].1;
+            let args = args(&hosts_ports, ports, Some(i));
```

Comment:
This looks exactly right, but unfortunately it is a good bit harder with @jkosh44's latest changes—you'll see when you rebase.

Reply:
Conceptually it's simple, no? You can still break apart the loop and precompute all the mappings, you just need to keep track of the ones you read the  ports out of a file for so you can avoid passing the corresponding machines to `supervise` in the second loop.

Reply:
Yep, definitely. It's just mechanically more difficult!